pkg/models/models.go

```diff
@@ -0,0 +1,187 @@
+package models
+
+import (
+	"context"
+	"time"
+)
+
+type Response struct {
+	// github specific
+	// These fields provide the page values for paginating through a set of
+	// results. Any or all of these may be set to the zero value for
+	// responses that are not part of a paginated set, or for which there
+	// are no additional pages.
+	//
+	// These fields support what is called "offset pagination" and should
+	// be used with the ListOptions struct.
+	NextPage  int
+	PrevPage  int
+	FirstPage int
+	LastPage  int
+
+	// Additionally, some APIs support "cursor pagination" instead of offset.
+	// This means that a token points directly to the next record which
+	// can lead to O(1) performance compared to O(n) performance provided
+	// by offset pagination.
+	//
+	// For APIs that support cursor pagination (such as
+	// TeamsService.ListIDPGroupsInOrganization), the following field
+	// will be populated to point to the next page.
+	//
+	// To use this token, set ListCursorOptions.Page to this value before
+	// calling the endpoint again.
+	NextPageToken string
+
+	// Explicitly specify the Rate type so Rate's String() receiver doesn't
+	// propagate to Response.
+	Rate Rate
+}
+
+type Repo struct {
+	Organization string
+	Project      string
+	Host         string
+	Group        string
+}
+
+type SearchParams struct {
+	Repo        Repo
+	Filters     []Filter
+	Ctx         context.Context
```

Comment:
Context should never be part of a struct: https://golang.org/pkg/context/

`Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. The Context should be the first parameter, typically named ctx:`

Reply:
Thanks for notice, fixed.